### PR TITLE
Ad/1405 fix for adding a duplicate email to provide warning message

### LINKED ISSUE #1405: Adding a duplicate email in "Domain managers" tab does not provide warning message
### Current Behavior

Task: manage a domain

When adding a new user in the "Domain managers" tab, if one adds an email that is already listed, it doesn't give a warning that you are adding an email that is already there. Just display a successful message.

### Expected Behavior

Display a message to inform user that the email address is already listed.

### Steps to Reproduce

When adding a new user in the "Domain managers" tab, if one adds an email that is already listed, it doesn't give a warning that you are adding an email that is already there. Just display a successful message.

### Environment

_No response_

### Additional Context

_No response_

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by asaki222**
## Ticket

Resolves #1405

All other changes require just a single approving review.-->

## Changes

- Moved the domain invitation query from `_make_invitation` method into the `_send_domain_invitation_email` method. The message displayed now depends on the invite's status.
- Added a message to be displayed if adding a user as a domain manager fails, provided the user already has an account.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/views/domain.py
Comment (abroddrick) [inline src/registrar/views/domain.py:848]: @Katherine-Osos want to review the content on this error message? Or is it mild enough you want to skip it?

File: src/registrar/views/domain.py
Comment (abroddrick) [inline src/registrar/views/domain.py:853]: Same here Katherine

**Comment by erinysong** [comment]
Confirming that the warning alert successfully displays when adding a duplicate email to domain managers
image

**Review by erinysong — APPROVED** [review]
Verified that alert successfully displays and code LGTM from a dev perspective - good job! I'm good with this being merged once we get content's approval

File: src/registrar/views/domain.py
Comment (Katherine-Osos) [reply src/registrar/views/domain.py:853]: This message looks good!

File: src/registrar/views/domain.py
Comment (Katherine-Osos) [reply src/registrar/views/domain.py:848]: This message looks good!

File: src/registrar/views/domain.py
Comment (Katherine-Osos) [inline src/registrar/views/domain.py:928]: @asaki222 This scenario seems similar to user already accepting a domain invitation (line 848 in the code). i.e. they're already a domain manager. Is that correct? 

If so, then I think the alert message should match that other scenario: 

> "{email} is already a manager for this domain."

Comment (asaki222) [reply src/registrar/views/domain.py]: @Katherine-Osos updated the text!

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

**Pull request merged**
